shipment_advice/models/shipment_advice.py

```diff
@@ -429,6 +430,25 @@ def _get_error_message(self, error, related_object):
             error=str(error),
         )
 
+    def auto_close_incoming_shipment_advices(self):
+        """Set incoming shipment advice to done when all planned moves are processed"""
+        if self.env.context.get("shipment_advice_ignore_auto_close"):
+            return
+        shipment_ids_to_close = []
+        for shipment in self:
+            if (
+                shipment.shipment_type != "incoming"
+                or not shipment.company_id.shipment_advice_auto_close_incoming
+                or any(
+                    move.state not in ("cancel", "done")
+                    for move in shipment.planned_move_ids
+                )
+            ):
+                continue
+            shipment_ids_to_close.append(shipment.id)
+        if shipment_ids_to_close:
+            self.browse(shipment_ids_to_close)._action_done()
```

Comment:
With https://github.com/OCA/stock-logistics-transport/pull/132 i refactored `_action_done` and moved all of the parts for validating transfers and create backorders into a own method. To just set the state and the departure date, because this is all what is needed, when called from `auto_close_incoming_shipment_advices`.

The same should be now possible by calling `_postprocess_action_done()` 
```suggestion
            self.browse(shipment_ids_to_close)._postprocess_action_done()
```

Reply:
done, thanks!

Reply:
Tests are now failing

Reply:
yes because here the state is to `in_process` 
https://github.com/OCA/stock-logistics-transport/blob/00ec7f151b31528fe5d4df887bd283dc85c4f8fc/shipment_advice/models/shipment_advice.py#L331
and here it is getting checked https://github.com/OCA/stock-logistics-transport/blob/00ec7f151b31528fe5d4df887bd283dc85c4f8fc/shipment_advice/models/shipment_advice.py#L407

Reply:
I added one more method `_close_shipments()` that updates only relevant fields, both for shipments to autoclose, and processed shipments through `_postprocess_action_done()`. Also, `_postprocess_action_done` has an `self.ensure_one()` check, while `_close_shipments()` supports a recordset.